app/Services/Locations/InternalRentalBillingService.php

```
@@ -0,0 +1,116 @@
+
+        $billingKey = sprintf('INT-%s-%s', $fixedAsset->id, $periodKey);
+
+        $existing = $this->existingInvoice($fixedAsset, $billingKey);
+        if ($existing) {
+            return $existing;
+        }
+
+        $days = $periodStart->copy()->startOfDay()->diffInDays($periodEnd->copy()->addDay()->startOfDay());
+        $amount = round($days * (float) $fixedAsset->daily_rate, 2);
+
+        return $fixedAsset->internalRentalInvoices()->create([
```

PHPStan (2.2.7)

[error] Method App\Services\Locations\InternalRentalBillingService::generateForAsset() should return App\Models\Locations\InternalRentalInvoice|null but returns Illuminate\Database\Eloquent\Model.


(return.type)